gunicorn/arbiter.py

```diff
@@ -460,10 +461,12 @@ def reap_workers(self):
                     if exitcode == self.APP_LOAD_ERROR:
                         reason = "App failed to load."
                         raise HaltServer(reason, self.APP_LOAD_ERROR)
-                    worker = self.WORKERS.pop(wpid, None)
-                    if not worker:
-                        continue
-                    worker.tmp.close()
+
+                    worker = (self.WORKERS.pop(wpid, None)
```

Comment:
No, `reap_workers` is only called in response to `SIGCHILD` as before.

`self.DYING_WORKERS` exists so that `kill` can remove pids from `self.WORKERS` while we're waiting for `SIGCHILD` to come in and be processed.